internal/charm/services/repofactory.go

```diff
@@ -44,28 +41,19 @@ func NewCharmRepoFactory(cfg CharmRepoFactoryConfig) *CharmRepoFactory {
 		logger:             cfg.Logger,
 		charmhubHTTPClient: cfg.CharmhubHTTPClient,
 		modelConfigService: cfg.ModelConfigService,
-		memoizedRepos:      make(map[corecharm.Source]corecharm.Repository),
 	}
 }
 
 // GetCharmRepository returns a suitable corecharm.Repository instance for the
 // requested source. Lookups are memoized for future requests.
 func (f *CharmRepoFactory) GetCharmRepository(ctx context.Context, src corecharm.Source) (corecharm.Repository, error) {
-	f.mu.Lock()
-	defer f.mu.Unlock()
-
-	if repo, isCached := f.memoizedRepos[src]; isCached {
-		return repo, nil
-	}
-
-	var repo corecharm.Repository
```

Comment:
If the model config changed, then you can't change the repo :man_facepalming: the only way to totally correct this would be to bounce the agent. What's worse it was inconsistent, as the "constructor" was called in the facade register, which meant that new facades were correct and the old ones weren't.

This wasn't holding its weight.